src/ControlSystem/WriteData.hpp

```diff
@@ -0,0 +1,174 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <cstddef>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "ControlSystem/Component.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "Domain/FunctionsOfTime/FunctionOfTime.hpp"
+#include "Domain/FunctionsOfTime/QuaternionFunctionOfTime.hpp"
+#include "IO/Observer/Helpers.hpp"
+#include "IO/Observer/ObservationId.hpp"
+#include "IO/Observer/ReductionActions.hpp"
+#include "IO/Observer/TypeOfObservation.hpp"
+#include "Parallel/GlobalCache.hpp"
+#include "Parallel/Info.hpp"
+#include "Parallel/Invoke.hpp"
+#include "Parallel/Reduction.hpp"
+#include "Utilities/Functional.hpp"
+#include "Utilities/MakeString.hpp"
+
+namespace control_system {
+namespace detail {
+struct WriterHelper {
+  // Use funcl::AssertEqual for all because we aren't actually "reducing"
+  // anything. These are just one time measurements.
+  using ReductionData = Parallel::ReductionData<
+      // Time
+      Parallel::ReductionDatum<double, funcl::AssertEqual<>>,
+      // Lambda, dtLambda, d2tLambda
+      Parallel::ReductionDatum<double, funcl::AssertEqual<>>,
+      Parallel::ReductionDatum<double, funcl::AssertEqual<>>,
+      Parallel::ReductionDatum<double, funcl::AssertEqual<>>,
+      // Q, dtQ, control signal
+      Parallel::ReductionDatum<double, funcl::AssertEqual<>>,
+      Parallel::ReductionDatum<double, funcl::AssertEqual<>>,
+      Parallel::ReductionDatum<double, funcl::AssertEqual<>>>;
+
+  using observed_reduction_data_tags =
+      observers::make_reduction_data_tags<tmpl::list<ReductionData>>;
+
+  const static inline std::vector<std::string> legend{
```

Comment:
Can you add a comment like `// All controlled function of time components have the same legend`